userspace/libsinsp/test/sinsp_utils.ut.cpp

```diff
@@ -19,122 +19,123 @@ limitations under the License.
-	EXPECT_EQ("/dir/.../term", std::string(target));
-
-	path = "/dir//...";
-	copy_and_sanitize_path(target, target, path.c_str(), '/');
-	EXPECT_EQ("/dir/...", std::string(target));
+	// Some tests were motivated by this resource:
+	// https://pubs.opengroup.org/onlinepubs/000095399/basedefs/xbd_chap04.html#tag_04_11
+
+	std::string path1, path2, res;
+
+	res = sinsp_utils::concatenate_paths("", "");
+	EXPECT_EQ("", res);
+
+	path1 = "";
+	path2 = "../";
+	res = sinsp_utils::concatenate_paths(path1, path2);
+	EXPECT_EQ("..", res);
+
+	path1 = "";
+	path2 = "./";
+	res = sinsp_utils::concatenate_paths(path1, path2);
+	EXPECT_EQ(".", res);
+
+	path1 = "";
+	path2 = "dir/term";
+	res = sinsp_utils::concatenate_paths(path1, path2);
+	EXPECT_EQ(path2, res);
+
+	path1 = "";
+	path2 = "//dir/term";
+	res = sinsp_utils::concatenate_paths(path1, path2);
+	EXPECT_EQ("/dir/term", res);
+
+	path1 = "/";
+	path2 = "dir/term";
+	res = sinsp_utils::concatenate_paths(path1, path2);
+	EXPECT_EQ("/dir/term", res);
+
+	path1 = "";
+	path2 = "///dir/term";
+	res = sinsp_utils::concatenate_paths(path1, path2);
+	EXPECT_EQ("/dir/term", res);
+
+	path1 = "";
+	path2 = "./dir/term";
+	res = sinsp_utils::concatenate_paths(path1, path2);
+	EXPECT_EQ("dir/term", res);
+
+	path1 = "/";
+	path2 = "//dir//////term";
+	res = sinsp_utils::concatenate_paths(path1, path2);
+	EXPECT_EQ("/dir/term", res);
+
+	path1 = "/";
+	path2 = "/dir/term";
+	res = sinsp_utils::concatenate_paths(path1, path2);
+	EXPECT_EQ("/dir/term", res);
+
+	path1 = "../.../";
+	path2 = "dir/././././../../.../term/";
+	res = sinsp_utils::concatenate_paths(path1, path2);
+	EXPECT_EQ("../.../term", res);
+
+	path1 = "../.../";
+	path2 = "/app/custom/dir/././././../../.../term/";
+	res = sinsp_utils::concatenate_paths(path1, path2);
+	EXPECT_EQ("/app/.../term", res);
+
+	path1 = "../.../";
+	path2 = "/app/custom/dir/././././../../term/";
+	res = sinsp_utils::concatenate_paths(path1, path2);
+	EXPECT_EQ("/app/term", res);
+
+	path1 = "./app";
+	path2 = "custom/term";
+	res = sinsp_utils::concatenate_paths(path1, path2);
+	EXPECT_EQ("app/custom/term", res);
+
+	path1 = "/app";
+	path2 = "custom/term";
+	res = sinsp_utils::concatenate_paths(path1, path2);
+	EXPECT_EQ("/app/custom/term", res);
+
+	path1 = "app";
+	path2 = "custom/term";
+	res = sinsp_utils::concatenate_paths(path1, path2);
+	EXPECT_EQ("app/custom/term", res);
+
+	path1 = "app//";
+	path2 = "custom/term";
+	res = sinsp_utils::concatenate_paths(path1, path2);
+	EXPECT_EQ("app/custom/term", res);
+
+	path1 = "app/////";
+	path2 = "custom////term";
+	res = sinsp_utils::concatenate_paths(path1, path2);
+	EXPECT_EQ("app/custom/term", res);
+
+	path1 = "/";
+	path2 = "/app/custom/dir/././././../../term/";
+	res = sinsp_utils::concatenate_paths(path1, path2);
+	EXPECT_EQ("/app/term", res);
+
+	path1 = "/";
+	path2 = "////app";
+	res = sinsp_utils::concatenate_paths(path1, path2);
+	EXPECT_EQ("/app", res);
+
+	path1 = "/root/";
+	path2 = "../😉";
```

Comment:
Fix https://github.com/falcosecurity/falco/issues/2958

Additional note: @mstemm `fs.path.*` also doesn't do any sanitation in case we are interested in equalizing `fd.name` and `fs.path.name` in the future.